src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -28,6 +28,346 @@ public static Vector<T> InitVector<T>(Func<int, T> f)
             return new Vector<T>(arr);
         }
 
+        public static byte[] CreateMaskForFirstActiveElement(byte[] mask, byte[] srcMask)
```

Comment:
why this can't be simple as:

```c#
public static byte[] CreateMaskForFirstActiveElement(byte[] mask, byte[] srcMask)
{
    for (var i = 0; i < count; i++)
    {
        // For active lane, set the element
        if (mask[i] != 0)
        {
            srcMask[i] = 1;
            break;
        }
    }
    return srcMask;
}
```

Reply:
Also, wondering if we can templatize it?

Reply:
We might be able to; but would need to do some reflection to dynamically handle zero and one for the primitives.

Reply:
It'll get templatized when this all gets copied back into `helpers.tt`

Reply:
I am ok to not do templatize version if it is too complicated. We already have lot of helpers that do not use template version.